src/components/UserAccounts/index.js

```diff
@@ -0,0 +1,118 @@
+    const { getUsersList } = this.props;
+    getUsersList();
+  }
+
+
+  render() {
+    const { users, isFetched, isRetrieving } = this.props;
+    const clusterName = localStorage.getItem('clusterName');
+    const { match: { params } } = this.props;
+
+    return (
+      <div className="MainPage">
+        <div className="TopBarSection"><Header /></div>
+        <div className="MainSection">
+          <div className="SideBarSection">
+            <SideNav clusterName={clusterName} clusterId={params.clusterID} />
+          </div>
+          <div className="MainContentSection">
+            <div className="InformationBarSection">
+              <InformationBar header="User Accounts" showBtn={false} />
+            </div>
+            <div className="ContentSection">
+              <div className="ResourcesTable">
+                <table className="UsersTable">
+                  <tr>
+                    <th>Name</th>
+                    <th>Email</th>
+                    <th>Role</th>
+                  </tr>
+                  {
+                    isRetrieving ? (
+                      <tr className="TableLoading">
+                        <div className="SpinnerWrapper">
+                          <BigSpinner />
+                        </div>
+                      </tr>
+                    ) : (
+                      <tbody>
+                        {isFetched && users !== undefined && (
+                          (users.map((user) => (
+                            <tr>
+                              <td>{user.name}</td>
+                              <td>{user.email}</td>
+                              <td>{user.roles[0].name}</td>
+                            </tr>
+                          )))
+                        )}
+                      </tbody>
+                    )
+                  }
+                </table>
+                {(isFetched && users.length === 0) && (
+                  <div className="NoContentDiv">
+                    <p>No Users Available</p>
+                  </div>
+                )}
+                {(!isRetrieving && !isFetched) && (
+                  <div className="NoContentDiv">
+                    <p>
+                      Oops! Something went wrong!
+
+                      Failed to retrieve Available Users.
+                    </p>
+                  </div>
+                )}
+
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+
+    );
+  }
+}
+
+// inititate props
+UserAccounts.propTypes = {
+  users: PropTypes.arrayOf(PropTypes.object),
+  isRetrieving: PropTypes.bool,
+  isFetched: PropTypes.bool,
+  getUsersList: PropTypes.func.isRequired
+};
+
+// assigning defaults
+UserAccounts.defaultProps = {
+  users: [],
+  isRetrieving: false,
+  isFetched: false,
+};
+
+export const mapStateToProps = (state) => {
+  const { isRetrieving, users, isFetched } = state.UsersListReducer;
+  return { isRetrieving, users, isFetched };
```

Comment:
Also for uniformity, when you use `isFetched`, use `isFetching` and when you use `isRetrieved` and use `isRetrieving` for the props values, this creates conflicts and errors now like on the projects page, it uses `isRetrieving` to get projects and the same page tries to use props for `UsersListReducer`, meaning `isRetrieving` for users will update `isRetrieving` for projects. Does this make sense?

Reply:
@Mubangizi cool